src/main/kotlin/com/mparticle/kits/StringUtils.kt

```diff
@@ -11,50 +8,45 @@ internal object StringUtils {
         defaultValue: Boolean
     ): Boolean {
         val value = settings[key]
-        if (value != null) {
-            val flag = value.toBoolean()
-            if (flag != null) {
-                return flag
-            }
-            ApptentiveLog.w(
-                ApptentiveLogTag.UTIL,
-                "Unable to parse boolean flag '%s': %s",
-                key,
-                value
-            )
+        return value?.toBoolean() ?: defaultValue
+    }
+
+    fun tryParseLongSettingFlag(
+        settings: Map<String, String>,
+        key: String?,
+        defaultValue: Long
+    ): Long {
+        val value = settings[key]
```

Comment:
We could wrap the block in key?.let { } since the Map doesnt have nullable keys

Reply:
Added the null check. Also, updated the tests with a null key case.